File: SDKs/zoom.us/2.0.0/python/src/sdk/models/operations/accountplanaddoncancel.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import security as shared_security


@dataclasses.dataclass
class AccountPlanAddonCancelPathParams:
    account_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'accountId', 'style': 'simple', 'explode': False }})
    
class AccountPlanAddonCancelApplicationJSONActionEnum(str, Enum):
    CANCEL = "cancel"


@dataclass_json
@dataclasses.dataclass
class AccountPlanAddonCancelApplicationJSON:
    r"""AccountPlanAddonCancelApplicationJSON
    Account plan object
    """
    
    action: Optional[AccountPlanAddonCancelApplicationJSONActionEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('action') }})
    comment: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('comment') }})
    reason: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('reason') }})
    type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
class AccountPlanAddonCancelMultipartFormDataActionEnum(str, Enum):
    CANCEL = "cancel"


@dataclasses.dataclass
class AccountPlanAddonCancelMultipartFormData:
    r"""AccountPlanAddonCancelMultipartFormData
    Account plan object
    """
    
    action: Optional[AccountPlanAddonCancelMultipartFormDataActionEnum] = dataclasses.field(default=None, metadata={'multipart_form': { 'field_name': 'action' }})
    comment: Optional[str] = dataclasses.field(default=None, metadata={'multipart_form': { 'field_name': 'comment' }})
    reason: Optional[int] = dataclasses.field(default=None, metadata={'multipart_form': { 'field_name': 'reason' }})
    type: Optional[str] = dataclasses.field(default=None, metadata={'multipart_form': { 'field_name': 'type' }})
    

@dataclasses.dataclass
class AccountPlanAddonCancelRequests:
    object: Optional[AccountPlanAddonCancelApplicationJSON] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    object1: Optional[AccountPlanAddonCancelMultipartFormData] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'multipart/form-data' }})
    

@dataclasses.dataclass
class AccountPlanAddonCancelSecurity:
    o_auth: shared_security.SchemeOAuth = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class AccountPlanAddonCancelRequest:
    path_params: AccountPlanAddonCancelPathParams = dataclasses.field()
    security: AccountPlanAddonCancelSecurity = dataclasses.field()
    request: Optional[AccountPlanAddonCancelRequests] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class AccountPlanAddonCancelResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    body: Optional[bytes] = dataclasses.field(default=None)
    account_plan_addon_cancel_204_application_json_any: Optional[Any] = dataclasses.field(default=None)
    

File: SDKs/zoom.us/2.0.0/typescript/src/sdk/models/operations/accountplanaddoncancel.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class AccountPlanAddonCancelPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=accountId" })
  accountId: string;
}

export enum AccountPlanAddonCancelApplicationJsonActionEnum {
    Cancel = "cancel"
}


// AccountPlanAddonCancelApplicationJson
/** 
 * Account plan object
**/
export class AccountPlanAddonCancelApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=action" })
  action?: AccountPlanAddonCancelApplicationJsonActionEnum;

  @SpeakeasyMetadata({ data: "json, name=comment" })
  comment?: string;

  @SpeakeasyMetadata({ data: "json, name=reason" })
  reason?: number;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}

export enum AccountPlanAddonCancelMultipartFormDataActionEnum {
    Cancel = "cancel"
}


// AccountPlanAddonCancelMultipartFormData
/** 
 * Account plan object
**/
export class AccountPlanAddonCancelMultipartFormData extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "multipart_form, name=action" })
  action?: AccountPlanAddonCancelMultipartFormDataActionEnum;

  @SpeakeasyMetadata({ data: "multipart_form, name=comment" })
  comment?: string;

  @SpeakeasyMetadata({ data: "multipart_form, name=reason" })
  reason?: number;

  @SpeakeasyMetadata({ data: "multipart_form, name=type" })
  type?: string;
}


export class AccountPlanAddonCancelRequests extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  object?: AccountPlanAddonCancelApplicationJson;

  @SpeakeasyMetadata({ data: "request, media_type=multipart/form-data" })
  object1?: AccountPlanAddonCancelMultipartFormData;
}


export class AccountPlanAddonCancelSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oAuth: shared.SchemeOAuth;
}


export class AccountPlanAddonCancelRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: AccountPlanAddonCancelPathParams;

  @SpeakeasyMetadata()
  request?: AccountPlanAddonCancelRequests;

  @SpeakeasyMetadata()
  security: AccountPlanAddonCancelSecurity;
}


export class AccountPlanAddonCancelResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  accountPlanAddonCancel204ApplicationJSONAny?: any;
}
